vaadin-time-picker-flow-parent/vaadin-time-picker-flow/src/main/resources/META-INF/resources/frontend/timepickerConnector.js

```diff
@@ -23,73 +32,6 @@
 
         timepicker.$connector = {};
 
-        const getAmPmString = function (locale, testTime) {
-          const testTimeString = testTime.toLocaleTimeString(locale);
-          // AM/PM string is anything from one letter in eastern arabic to standard two letters,
-          // to having space in between, dots ...
-          // cannot disqualify whitespace since some locales use a. m. / p. m.
-          // TODO when more scripts support is added (than Arabic), need to exclude those numbers too
-          const endWithAmPmRegex = /[^\d\u0660-\u0669]+$/g;
-          let amPmString = testTimeString.match(endWithAmPmRegex);
-          if (!amPmString) {
-            // eg. chinese (and some else too) starts with am/pm
-            amPmString = testTimeString.match(/^[^\d\u0660-\u0669]+/g);
-          }
-          if (amPmString) {
-            amPmString = amPmString[0].trim();
-          }
-          return amPmString;
-        };
-        const testPmTime = new Date('August 19, 1975 23:15:30');
-        const testAmTime = new Date('August 19, 1975 05:15:30');
-
-        const getPmString = function (locale) {
-          return getAmPmString(locale, testPmTime);
-        };
-        const getAmString = function (locale) {
-          return getAmPmString(locale, testAmTime);
-        };
-
-        // map from unicode eastern arabic number characters to arabic numbers
-        const arabicDigitMap = {
-          '\\u0660': '0',
-          '\\u0661': '1',
-          '\\u0662': '2',
-          '\\u0663': '3',
-          '\\u0664': '4',
-          '\\u0665': '5',
-          '\\u0666': '6',
-          '\\u0667': '7',
-          '\\u0668': '8',
-          '\\u0669': '9'
-        };
-
-        // parses eastern arabic number characters to arabic numbers (0-9)
-        const anyNumberCharToArabicNumberReplacer = function (charsToReplace) {
-          return charsToReplace.replace(/[\u0660-\u0669]/g, function (char) {
-            const unicode = '\\u0' + char.charCodeAt(0).toString(16);
-            return arabicDigitMap[unicode];
-          });
-        };
-
-        const parseAnyCharsToInt = function (anyNumberChars) {
-          return parseInt(anyNumberCharToArabicNumberReplacer(anyNumberChars));
-        };
-
-        const parseMillisecondCharsToInt = function (millisecondChars) {
-          millisecondChars = anyNumberCharToArabicNumberReplacer(millisecondChars);
-          // digits are either .1 .01 or .001 so need to "shift"
-          if (millisecondChars.length === 1) {
-            millisecondChars += '00';
-          } else if (millisecondChars.length === 2) {
-            millisecondChars += '0';
-          }
-          return parseInt(millisecondChars);
-        };
-
-        // detecting milliseconds from input, expects am/pm removed from end, eg. .0 or .00 or .000
-        const millisecondRegExp = /[[\.][\d\u0660-\u0669]{1,3}$/;
```

Comment:
**note:** This variable was moved [below](https://github.com/vaadin/flow-components/pull/3049/files?diff=split&w=1#diff-fd7ef2beaf5cba9e6bbd9ac707c7d34baec9fe9666353d069e9b20b6de8e5b53R136-R137), closer to the line where it is actually used.